numpyro/distributions/continuous.py

```diff
@@ -1033,6 +1033,115 @@ def icdf(self, q):
         return self.loc + self.scale * logit(q)
 
 
+def _kron(A, B):
+    D = A[..., :, None, :, None] * B[..., None, :, None, :]
+    ds = D.shape
+    newshape = (*ds[:-4], ds[-4] * ds[-3], ds[-2] * ds[-1])
+    return D.reshape(newshape)
+
+
+class MatrixNormal(Distribution):
+    arg_constraints = {
+        "loc": constraints.real_vector,
+        "scale_columns": constraints.positive_definite,
+        "scale_rows": constraints.positive_definite,
+    }
+    support = constraints.real_matrix
+    reparametrized_params = ["loc", "scale_columns", "scale_rows"]
+
+    def __init__(self, loc, scale_columns, scale_rows, validate_args=None):
+        """_summary_
+
+        Args:
+            loc (_type_): _description_
+            scale_columns (_type_): _description_
+            scale_rows (_type_): _description_
+            validate_args (_type_, optional): _description_. Defaults to None.
+
+        Raises:
+            ValueError: _description_
+        """
+        if not (loc.ndim == scale_columns.ndim == scale_rows.ndim):
+            raise ValueError(
+                "ndim of loc, scale_columns and scale_rows are reuqired to be equal."
+            )
+
+        event_shape = loc.shape[-2:]
+        n, p = event_shape
+        batch_shape = loc.shape[:-2]
+
+        assert scale_rows.shape == batch_shape + (
+            n,
+            n,
+        ), "scale_rows.shape does not match"
+        assert scale_columns.shape == batch_shape + (
+            p,
+            p,
+        ), "scale_rows.shape does not match"
+
+        super(MatrixNormal, self).__init__(
+            batch_shape=batch_shape, event_shape=event_shape
+        )
+
+        self.loc = loc
+        self.scale_columns = scale_columns
+        self.scale_rows = scale_rows
+
+    @property
+    def mean(self):
+        return self.loc
```

Comment:
We need broadcasting [like this](https://github.com/pyro-ppl/numpyro/blob/6fdee9d126d9857f071318614988ccccf09bac20/numpyro/distributions/continuous.py#L1273)

Reply:
Doesn't  `_promote_shapes` take care of this?

Reply:
Promoting shapes only adds singleton dimensions to the input.